containers/Dockerfile.buildenv

```diff
@@ -354,21 +354,50 @@ RUN apt-get -y clean
 WORKDIR /work
 
 
-# We don't inherit from ci because that has lots of unnecessary compilers that
-# we don't need and would only increase the size of the image. We inherit from
-# the remote image rather than the local dev because it is faster on release
-# CI to just pull the remote dev image and build demo rather than having to
-# build all of dev and demo combined.
-# Instead of basing demo off the remote dev image, we could pre-fetch the dev
-# image and then build from the local cache like
-# FROM dev AS demo
-# Either way works, we just chose to build from remote instead.
-FROM sxscollaboration/spectre:dev AS demo
+# Deploy compiled executables to an image that can be run on HPC systems.
+# - We could also compile in the dev container and then copy the executables to
+#   a minimal deploy container to reduce its size. That requires keeping track
+#   of the shared libraries that are needed to run the executables, so to make
+#   things easier we just base the deploy container on the dev container.
+# - We inherit from the remote image rather than the local dev because it is
+#   faster on release CI to just pull the remote dev image rather than having to
+#   build the dev container again. We could pre-fetch the dev image and then
+#   build from the local cache, but that didn't work right away.
+FROM sxscollaboration/spectre:dev AS deploy
```

Comment:
Are we going to have different `deploy` containers for different executables? Or just one all executables we'll offer? I could also see this container being called `cce` instead so it's more clear that's what it is.

Reply:
I thought we go with one deploy container for now to keep things simple. If it turns out the container increases in size too much we can split it up.